common/utils/query_utils.ts

```diff
@@ -9,10 +9,4 @@
  * GitHub history for details.
  */
 
-export const getIndexPatternFromRawQuery = (query: string) : string => {
-  const indexMatches = query.match('\=(.*)');
-  if (indexMatches) {
-    return indexMatches[0]?.slice(1)?.split('|')[0]?.trim();
-  }
-  return '';
-};
\ No newline at end of file
+export const getIndexPatternFromRawQuery = (query: string) : string => query.match(/(source|index)\s*=\s*([^|\s]+)/i)[1];
```

Comment:
i think the captured group index should be 2? since you added another one in front. Also maybe add some guard if no match? `query.match(...)?.[2] ?? ''` (or use a variable and add null check, i don't know does `?.` make regex match run one more time?)

Reply:
Yea it should be [2] but interesting though is that the app is running correctly with match...tried to reboot but still...and after the change it is also working. Anyway I changed it to use the [2] for matching, thanks for catching this. Yea, I think if there's no match the returned res would be null.

Reply:
hmm interesting, [1] should not work, and query.match() returns null if there is no match and indexing null will crash the application. your browser cache might be outdated, try "empty cache and hard reload" or incognito on browser?

Reply:
yea cache issue probably, now it's broken. but I used 'empty cache and hard reload' all the time...hmmm